static/app/views/settings/project/server-side-sampling/modals/uniformRateModal.tsx

```diff
@@ -209,12 +234,63 @@ function UniformRateModal({
   }
 
   function handleReadDocs() {
+    onReadDocs();
+
+    if (activeStep === undefined) {
+      return;
+    }
+
     trackAdvancedAnalyticsEvent('sampling.settings.modal.uniform.rate_read_docs', {
       organization,
       project_id: project.id,
     });
+  }
 
-    onReadDocs();
+  if (activeStep === undefined || loading) {
+    return (
+      <Fragment>
+        <Header closeButton>
+          <Placeholder height="22px" />
+        </Header>
+        <Body>
+          <LoadingIndicator />
+        </Body>
+        <Footer>
+          <FooterActions>
+            <Button
+              href={SERVER_SIDE_SAMPLING_DOC_LINK}
+              onClick={handleReadDocs}
+              external
+            >
+              {t('Read Docs')}
+            </Button>
+            <ButtonBar gap={1}>
+              <Button onClick={closeModal}>{t('Cancel')}</Button>
+              <Placeholder height="40px" width="80px" />
+            </ButtonBar>
+          </FooterActions>
+        </Footer>
+      </Fragment>
+    );
+  }
+
+  if (activeStep === Step.SET_CURRENT_CLIENT_SAMPLE_RATE) {
+    return (
+      <SpecifyClientRateModal
+        {...props}
+        Header={Header}
+        Body={Body}
+        Footer={Footer}
+        closeModal={closeModal}
+        onReadDocs={onReadDocs}
+        organization={organization}
+        projectId={project.id}
+        onGoNext={value => {
+          setSpecifiedClientRate(Number(value));
+          setActiveStep(Step.SET_UNIFORM_SAMPLE_RATE);
+        }}
+      />
+    );
```

Comment:
I've noticed one more bug.
If we hit back on the uniform rate modal the client rate is lost.
How about we keep it in a state of the uniformRateModal and pass it as props to the SpecifyClientRateModal (and updater function too)? We could remove the useState from the SpecifyClientRateModal and the value would be remembered when going back and forth between modals.

Reply:
good catch...thanks